components/popover_list_members/popover_list_members.tsx

```diff
@@ -0,0 +1,274 @@
+            status={props.statuses[user.id]}
+            user={user}
+        />
+    ));
+
+    const channelIsArchived = props.channel.delete_at !== 0;
+    let popoverButton = null;
+    let editButton = null;
+    if (props.channel.type !== Constants.GM_CHANNEL && !channelIsArchived) {
+        const isDefaultChannel = props.channel.name === Constants.DEFAULT_CHANNEL;
+        let membersName = (isDefaultChannel || !props.manageMembers) ? (
+            <FormattedMessage
+                id='members_popover.viewMembers'
+                defaultMessage='View Members'
+            />
+        ) : (
+            <FormattedMessage
+                id='members_popover.manageMembers'
+                defaultMessage='Manage Members'
+            />
+        );
+        if (props.manageMembers) {
+            if (props.addMembersABTest === AddMembersToChanneltreatments.BOTTOM) {
+                membersName = (
+                    <FormattedMessage
+                        id='members_popover.addMembers'
+                        defaultMessage='Add Members'
+                    />
+                );
+                editButton = (
+                    <button
+                        className='btn btn-link'
+                        id='editBtn'
+                        onClick={showMembersModal}
+                    >
+                        <i className='icon icon-pencil-outline'/>
+                        <FormattedMessage
+                            id='members_popover.editBtn'
+                            defaultMessage='Edit'
+                        />
+                    </button>
+                );
+            } else if (props.addMembersABTest === AddMembersToChanneltreatments.TOP) {
+                editButton = (
+                    <button
+                        className='btn btn-link'
+                        id='addBtn'
+                        onClick={getOnAddNewMembersButton(AddMembersToChanneltreatments.TOP)}
+                    >
+                        <i className='icon icon-account-plus-outline'/>
+                        <FormattedMessage
+                            id='members_popover.add'
+                            defaultMessage='Add'
+                        />
+                    </button>
+                );
+            }
+        }
+
+        const handleButtonOnClick = (props.manageMembers && props.addMembersABTest === AddMembersToChanneltreatments.BOTTOM) ? getOnAddNewMembersButton(AddMembersToChanneltreatments.BOTTOM) : showMembersModal;
+        popoverButton = (
+            <div
+                className='more-modal__button'
+                key={'popover-member-more'}
+            >
+                <button
+                    className='btn btn-link'
+                    data-testid='membersModal'
+                    onClick={handleButtonOnClick}
+                >
+                    {membersName}
+                </button>
+            </div>
+        );
+    }
```

Comment:
Thanks for the suggestion @michelengelen, with two separate functions it is well organized. I committed the changes. Please see my minor comments below. 
- For `renderHeaderButton`, I also thought about using ternary operator. Then I decided to keep as it is (i.e. check explicitly with if `props.addMembersABTest` is `top` or `bottom`. Because (maybe) in some cases `props.addMembersABTest` is something other than `top` or `bottom`. Anyhow, I assume it is safe to just check if it is `top` with ternary operator  
-  For `renderFooterButton`, it is a good idea just to use switch statement. There was a small issue related to one case where  
     - `isDefaultChannel` is `true`
     - `props.manageMembers` is `true`
     -  `props.addMembersABTest === AddMembersToChanneltreatments.BOTTOM`

   In this case `label` should be `Add Members` not `View Members`. I fixed it by changing the order and including `props.manageMembers` in the first case.